src/main/java/exchange/core2/core/orderbook/OrderBookNaiveImpl.java

```diff
@@ -321,6 +322,65 @@ public CommandResultCode cancelOrder(OrderCommand cmd) {
         return CommandResultCode.SUCCESS;
     }
 
+    /**
+     * Remove an order.<p>
+     *
+     * @param order cancel (order to remove)
+     * @return true if order removed, false if not found (can be removed/matched earlier)
+     */
+    public boolean cancelOrder(Order order) {
+        final long orderId = order.orderId;
+
+        if (order == null) {
+            // order already matched and removed from order book previously
+            return false;
+        }
+
+        // now can remove it
+        idMap.remove(orderId);
+
+        final NavigableMap<Long, OrdersBucketNaive> buckets = getBucketsByAction(order.action);
+        final long price = order.price;
+        final OrdersBucketNaive ordersBucket = buckets.get(price);
+        if (ordersBucket == null) {
+            // not possible state
+            throw new IllegalStateException("Can not find bucket for order price=" + price + " for order " + order);
+        }
+
+        // remove order and whole bucket if its empty
+        ordersBucket.remove(orderId, order.uid);
+        if (ordersBucket.getTotalVolume() == 0) {
+            buckets.remove(price);
+        }
+
+        return true;
+    }
+
+    /**
+     * Remove User orders.<p>
+     *
+     * @param cmd cancel orders (by UID)
+     * @return true if order removed, false if not found (can be removed/matched earlier)
+     */
+    public CommandResultCode cancelOrdersByUid(OrderCommand cmd) {
+        final long userId = cmd.uid;
+        AtomicInteger sizeOrders = new AtomicInteger(0);
```

Comment:
AtomicXxxx operations are very slow. You can use MutableInteger from Agrona. Or even just variable, if use classic for-each loop.

Reply:
Done